### ISSUE TITLE
proposal: "30 Days of #CodetoRust" Challenge - An extension of RainofRust Campaign

### ISSUE THREAD

**Issue by DineshMv** [MozillaIndia/RustIndia#13]
Hi there, 

I'd like to propose an extension to #1 . The whole idea is to initiate a "30 Days of #CodetoRust" challenge which is very similar to #100daysofcode campaign by freecodecamp except the timeline and rules are strictly restricted to Rust community. 

Please check this [**repo**](https://github.com/DineshMv/30Days-of-CodetoRust) to find out more.

Why this #CodetoRust challenge?

* After organizing an event, attendees will be asked to take up this challenge where each of them must learn/code something to rust lang everyday for an hour. 
* They must update the [log.md](https://github.com/DineshMv/30Days-of-CodetoRust/blob/master/Log.md) with the status of learning each day. It might be a link to code/screen shot/github link.
* We can track the no. of people who are interested to learn/contribute after the event. 
* we can track the projects/learning from every individual.
* People who contributed this challenge can help others to learn/code for rust. 

PS: The ultimate goal is to make people learn something about rust in 30days of time which is probably a good time for developers/individuals to continue their contributions and explore more about rust.

CC: @dvigneshwer @rowdymehul @couci @jayesh-kr

**Comment by dvigneshwer** [comment]
Great idea @DineshMv 

Really appreciate your enthusiasm and support to the campaign.

We should definitely try this in the second cycle, as we can't afford to add this in the middle of the campaign as a lot of planning needs to be done on this.

Here are my two cents on the subject,

* Let's try to maintain a single repo
** Give a PR to the RustIndia repo with a new directory containing the required files and update the Readme.md file accordingly 

* Regarding the new hashtag @rowdymehul can you add your POVs?
** Main aim is to track the social engagement better if new hashtag does the job I am super fine with it

* Don't you think it will be better if people write blogs instead of the log file so that it's more visible
** People should be encouraged to give PR of their work to the RustIndia repo so that we can create more Kits here
** This can help in tracking also

The main reason why I am pushing to use the RustIndia for collaboration is that we can have more conversations, content management, and support here.

**Comment by DineshMv** [comment]
>We should definitely try this in the second cycle, as we can't afford to add this in the middle of the campaign as a lot of planning needs to be done on this.

Sounds good. 

- >Give a PR to the RustIndia repo with a new directory containing the required files and update the Readme.md file accordingly

I will do this soon if I got some time. Still need to refine the draft.

>Don't you think it will be better if people write blogs instead of the log file so that it's more visible

Blog posts is too mainstream and for learning very small part of rust, writing a blogpost doesn't workout much. 
>- People should be encouraged to give PR of their work to the RustIndia repo so that we can create more Kits here

I like this idea of submitting PR's - we can create a page where every individual who take up this challenge will be asked to enter their names and date of starting like wise. Also, we can have page where people who completed 1week/10days of challenge to submit their works via hyperlinking the log file. This way people can give 3-4 PR's about their activity in this challenge. 

@dvigneshwer : Let me know your thoughts on this. 
PS: I am still waiting for @rowdymehul 's response on this issue in order to improve this. :)

**Comment by dvigneshwer** [comment]
Yeah, I think it's shaping better now but still, needs a lot of thoughts and feedbacks from others.

Will invite you in the phase 2 planning calls which will be set up soon after the analysis of first campaign results so that we can settle up with one implementation of the continuous learning program.

**Comment by DineshMv** [comment]
>Will invite you in the phase 2 planning calls which will be set up soon after the analysis of first campaign results so that we can settle up with one implementation of the continuous learning program.

Wow! I would love to join and will be awaiting to see this happening in phase 2 :)
